### PR TITLE
Added snippets for Logical and Bitwise operators in C++ along with th…

### PR DESCRIPTION
**Pull request by jatindhankhar**
Added snippets for Logical and Bitwise operators, inspired by the [Ruby section of Learnxinyminutes](http://learnxinyminutes.com/docs/ruby/)  along with alternate equivalent keywords for the operators.

### CODE REVIEW

**Comment by vendethiel** [comment]
Sorry for the delay. There appears to be a merge conflict with this file, I'd galdly merge it if you fix it :).

**Comment by vendethiel** [comment]
seems like the merge conflict is still present :(

**Comment by jatindhankhar** [comment]
Hi,
Sorry for the conflict issue. Seems like syncing fork isn't enough. 
Issue I am facing is similar to the OP of this [SO question](http://stackoverflow.com/questions/32030779/git-pull-request-says-this-branch-has-conflicts-that-must-be-resolved)
```bash
What I did was
Sync the fork
Pull from origin
Add and commit the changes
Then merge the upstream with local master
```


Am I missing something ? Should I create a new branch. Little confused

**Comment by vendethiel** [comment]
is "origin" your fork, or this repo?

Can you `git remote -v`?

**Comment by jatindhankhar** [comment]
Origin is the fork and upstream points to this repo

`git remote -v`

gives

```bash
origin	git@github.com:jatindhankhar/learnxinyminutes-docs.git (fetch)
origin	git@github.com:jatindhankhar/learnxinyminutes-docs.git (push)
upstream	https://github.com/adambard/learnxinyminutes-docs (fetch)
upstream	https://github.com/adambard/learnxinyminutes-docs (push)
```

image

**Comment by vendethiel** [comment]
Try

```
git fetch upstream
git merge upstream/master
```

**Comment by jatindhankhar** [comment]
Yes Did
Fetch succeeded
Merge has issues, same what Github was telling. 

```bash
Auto-merging zh-cn/bf-cn.html.markdown
Auto-merging tr-tr/bf-tr.html.markdown
Auto-merging ta_in/xml-ta.html.markdown
Auto-merging ta_in/javascript-ta.html.markdown
Auto-merging ta_in/css-ta.html.markdown
Auto-merging ru-ru/bf.html.markdown
Auto-merging pt-br/bf.html.markdown
Auto-merging pl-pl/bf-pl.html.markdown
Auto-merging nl-nl/bf.html.markdown
Auto-merging ko-kr/bf-kr.html.markdown
Auto-merging it-it/bf-it.html.markdown
Auto-merging hu-hu/ruby-hu.html.markdown
Auto-merging fr-fr/bf-fr.html.markdown
Auto-merging fa-ir/bf-fa.html.markdown
Auto-merging es-es/bf-es.html.markdown
Auto-merging c++.html.markdown
CONFLICT (content): Merge conflict in c++.html.markdown
Auto-merging bf.html.markdown
Automatic merge failed; fix conflicts and then commit the result.
```

Output of `git difftool`

```bash
 ❯ git difftool                                                      [13:47:40]
diff --cc c++.html.markdown
index 9d4d668,6b81f95..0000000
mode 100755,100644..100755
--- a/c++.html.markdown
+++ b/c++.html.markdown
@@@ -6,7 -6,7 +6,11 @@@ contributors
      - ["Matt Kline", "https://github.com/mrkline"]
      - ["Geoff Liu", "http://geoffliu.me"]
      - ["Connor Waters", "http://github.com/connorwaters"]
++<<<<<<< HEAD
 +    - ["Jatin Dhankhar, "https://github.com/jatindhankhar"]
++=======
+     - ["Ankush Goyal", "http://github.com/ankushg07"]
++>>>>>>> upstream/master
  lang: en
  ---
  
@@@ -802,47 -802,94 +806,138 @@@ void doSomethingWithAFile(const std::st
  //   all automatically destroy their contents when they fall out of scope.
  // - Mutexes using lock_guard and unique_lock
  
++<<<<<<< HEAD
 +///////////////////////////////////
 +// Logical and Bitwise operators
 +//////////////////////////////////
 +
 +//Most of the operators in C++ are same as in other languages
 +
 +// Logical operators
 +
 +true && false // Performs **logical and** to yield false
 +true || false // Performs **logical or** to yield true
 +! true // Performs **logcical not** to yield
 +
 +//Instead of using symbols equivalent keywords can be used
 +true and false // Performs **logical and** to yield false
 +true or false // Performs **logical or** to yield true
 +not true // Performs **logcical not** to yield
 +
 +//Bitwise operators
 +
 +// **<<** Left Shift Operator
 +// << shifts bits to the left
 +4 << 1 // Shifts bits of 4 to left by 1 to give 8
 +// x << n can be thought as x * 2^n
 +
 +
 +// **>>** Right Shift Operator
 +// << shifts bits to the right
 +4 >> 1 // Shifts bits of 4 to right by 1 to give 2
 +// x << n can be thought as x / 2^n
 +
 +~4 // Performs a bitwise not
 +4 | 3 // Performs bitwise or
 +4 & 3 // Performs bitwise and
 +4 ^ 3 // Performs bitwise xor
 +
 +//Equivalent keywords are
 +compl 4 // Performs a bitwise not
 +4 bitor 3 // Performs bitwise or
 +4 bitand 3 // Performs bitwise and   
 +4 xor 3 // Performs bitwise xor
 +
++=======
+ // containers with object keys of non-primitive values (custom classes) require
+ // compare function in the object itself or as a function pointer. Primitives
+ // have default comparators, but you can override it.
+ class Foo {
+ public:
+ 	int j;
+ 	Foo(int a) : j(a) {}
+ };
+ struct compareFunction {
+     bool operator()(const Foo& a, const Foo& b) const {
+         return a.j < b.j;
+     }
+ };
+ //this isn't allowed (although it can vary depending on compiler)
+ //std::map<Foo, int> fooMap;
+ std::map<Foo, int, compareFunction> fooMap;
+ fooMap[Foo(1)]  = 1;
+ fooMap.find(Foo(1)); //true
+ 
+ ///////////////////////////////////////
+ // Lambda Expressions (C++11 and above)
+ ///////////////////////////////////////
+ 
+ // lambdas are a convenient way of defining an anonymous function
+ // object right at the location where it is invoked or passed as
+ // an argument to a function.
+ 
+ // For example, consider sorting a vector of pairs using the second
+ // value of the pair
+ 
+ vector<pair<int, int> > tester;
+ tester.push_back(make_pair(3, 6));
+ tester.push_back(make_pair(1, 9));
+ tester.push_back(make_pair(5, 0));
+ 
+ // Pass a lambda expression as third argument to the sort function
+ // sort is from the <algorithm> header
+ 
+ sort(tester.begin(), tester.end(), [](const pair<int, int>& lhs, const pair<int, int>& rhs) {
+         return lhs.second < rhs.second;
+     });
+ 
+ // Notice the syntax of the lambda expression,
+ // [] in the lambda is used to "capture" variables
+ // The "Capture List" defines what from the outside of the lambda should be available inside the function body and how.
+ // It can be either:
+ //     1. a value : [x]
+ //     2. a reference : [&x]
+ //     3. any variable currently in scope by reference [&]
+ //     4. same as 3, but by value [=]
+ // Example:
+ 
+ vector<int> dog_ids;
+ // number_of_dogs = 3;
+ for(int i = 0; i < 3; i++) {
+ 	dog_ids.push_back(i);
+ }
+ 
+ int weight[3] = {30, 50, 10};
+ 
+ // Say you want to sort dog_ids according to the dogs' weights
+ // So dog_ids should in the end become: [2, 0, 1]
+ 
+ // Here's where lambda expressions come in handy
+ 
+ sort(dog_ids.begin(), dog_ids.end(), [&weight](const int &lhs, const int &rhs) {
+         return weight[lhs] < weight[rhs];
+     });
+ // Note we captured "weight" by reference in the above example.
+ // More on Lambdas in C++ : http://stackoverflow.com/questions/7627098/what-is-a-lambda-expression-in-c11
+ 
+ ///////////////////////////////
+ // Range For (C++11 and above)
+ ///////////////////////////////
+ 
+ // You can use a range for loop to iterate over a container
+ int arr[] = {1, 10, 3};
+ 
+ for(int elem: arr){
+ 	cout << elem << endl;
+ }
+ 
+ // You can use "auto" and not worry about the type of the elements of the container
+ // For example:
+ 
+ for(auto elem: arr) {
+ 	// Do something with each element of arr
+ }
++>>>>>>> upstream/master
  
  /////////////////////
  // Fun stuff
```

**Comment by vendethiel** [comment]
I've never used mergetool, bu tit seems to me you're supposed to pick from left side (yours) and right side (our master) to create the "fixed" version.

**Comment by jatindhankhar** [comment]
I think, this should work :sweat_smile:

File: c++.html.markdown [hunk 1]
```diff
@@ -6,7 +6,7 @@ contributors:
     - ["Matt Kline", "https://github.com/mrkline"]
     - ["Geoff Liu", "http://geoffliu.me"]
     - ["Connor Waters", "http://github.com/connorwaters"]
-    - ["Ankush Goyal", "http://github.com/ankushg07"]
+    - ["Jatin Dhankhar, "https://github.com/jatindhankhar"]
```
Comment (vendethiel) [inline c++.html.markdown]: that's not very nice :P

File: c++.html.markdown [hunk 2]
```diff
@@ -802,94 +802,6 @@ void doSomethingWithAFile(const std::string& filename)
 //   all automatically destroy their contents when they fall out of scope.
 // - Mutexes using lock_guard and unique_lock
 
-// containers with object keys of non-primitive values (custom classes) require
```
Comment (vendethiel) [inline c++.html.markdown]: uhm.

**Comment by jatindhankhar** [comment]
I think I messed up.
Thinking of doing this now
Replace contents of this file with fresh copy from master and then add the changes .

**Comment by vendethiel** [comment]
Please do that, yes :).

I think you should open another PR. I'm going to close this one in the meantime.

**Comment by jatindhankhar** [comment]
Yes 
Cool

**Pull request closed without merging**